src/imitation/algorithms/adversarial/common.py

```diff
@@ -421,7 +422,7 @@ def train_gen(
     def train(
         self,
         total_timesteps: int,
-        callback: Optional[Callable[[int], None]] = None,
+        callback: Optional[List[BaseCallback]] = None
```

Comment:
Do we want to change the semantics of the argument here, or should we rather deprecate the feature (and introduce a different parameter for additional gen_callback)?

I think the suggestion in the original issue was to add a new `gen_callback` argument. (Btw, stable-baselines [supports both](https://stable-baselines3.readthedocs.io/en/master/guide/callbacks.html#callbacklist) CallbackList and list of callbacks if we wanted to be fancy)